packages/react-log-viewer/src/LogViewer/LogViewer.tsx

```diff
@@ -108,40 +114,63 @@ export const LogViewer: React.FunctionComponent<LogViewerProps> = memo(
         highlightedRowIndexes,
         setHighlightedRowIndexes
       }),
-      [parsedData, logViewerRef, rowInFocus, searchedWordIndexes, highlightedRowIndexes, setHighlightedRowIndexes]
+      [data, parsedData, logViewerRef, rowInFocus, searchedWordIndexes, highlightedRowIndexes, setHighlightedRowIndexes]
     );
 
-    /* Parsing our data immediately after first component render */
+    /* Parse data every time it changes */
     useEffect(() => {
       setParsedData(parseConsoleOutput(data));
     }, [data]);
 
+    useEffect(() => {
+      if (logViewerRef && logViewerRef.current) {
+        logViewerRef.current.resetAfterIndex(0);
+      }
+    }, [parsedData]);
+
+    useEffect(() => {
+      if (scrollToRow && parsedData.length) {
+        setRowInFocus(parsedData.length - 1);
+        if (logViewerRef && logViewerRef.current) {
+          logViewerRef.current.scrollToItem(scrollToRow, 'center');
+        }
+      }
+    }, [parsedData, scrollToRow]);
+
     /* Updating searchedResults context state given changes in searched input */
     useEffect(() => {
       let foundKeywordIndexes: (number | null)[] = [];
       const adjustedSearchedInput = escapeString(searchedInput);
 
       if (adjustedSearchedInput !== '' && adjustedSearchedInput.length >= 3) {
-        foundKeywordIndexes = searchForKeyword(adjustedSearchedInput, parsedData);
+        foundKeywordIndexes = searchForKeyword(adjustedSearchedInput, parsedData, itemCount || parsedData.length - 1);
 
         if (foundKeywordIndexes.length !== 0) {
           setSearchedWordIndexes(foundKeywordIndexes);
-          scrollToRow(foundKeywordIndexes[DEFAULT_SEARCH_INDEX]);
+          scrollToRowInFocus(foundKeywordIndexes[DEFAULT_SEARCH_INDEX]);
         }
       }
 
       if (adjustedSearchedInput !== '' && adjustedSearchedInput.length < 3) {
-        setRowInFocus(DEFAULT_FOCUS);
+        if (scrollToRow) {
+          setRowInFocus(scrollToRow);
+        } else {
+          setRowInFocus(DEFAULT_FOCUS);
+        }
```

Comment:
Isn't this possible? 🤔 Can't find `DEFAULT_FOCUS` definition, so I'm not sure.

```suggestion
        setRowInFocus(scrollToRow || DEFAULT_FOCUS);
```